cpp/src/arrow/compute/kernels/scalar_boolean.cc

```diff
@@ -175,15 +175,24 @@ struct KleeneAnd : Commutative<KleeneAnd> {
     }
 
     if (right_true) {
-      GetBitmap(*out, 0).CopyFrom(GetBitmap(left, 0));
+      if (left.null_count == 0) {
```

Comment:
Should use `GetNullCount()`?
I see `null_count` is used extensively in this file.

Reply:
Yes, `GetNullCount()` will cache the null count value so that it's only computed once.